devito/interfaces.py

```diff
@@ -429,7 +452,20 @@ def _allocate_memory(self):
         self._full_data = self._data.view()
 
         if self.pad_time:
-            self._data = self._data[self.time_order:, :, :]
+            self._data = self._data[self.time_order:, ...]
+
+    def indexify(self):
+        """Convert base symbol and dimensions to indexed data accesses
+
```

Comment:
>Everything else is unintended "magic behaviour from a user's perspective.

Still better than non-working and wrong magic behaviour. Current status time handling is bad at best and need to be fixed.